service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsController.java

```diff
@@ -116,21 +118,24 @@ public void handleSubmittedEvent(
         CaseDetails caseDetails = callbackRequest.getCaseDetails();
         CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
 
-        if (displayAmountToPay(caseDetails)) {
-            try {
-                paymentService.makePaymentForC2(caseDetails.getId(), caseData);
-            } catch (FeeRegisterException | PaymentsApiException ignore) {
-                applicationEventPublisher.publishEvent(new FailedPBAPaymentEvent(callbackRequest, C2_APPLICATION));
-            }
-        } else if (NO.getValue().equals(caseDetails.getData().get(DISPLAY_AMOUNT_TO_PAY))) {
-            applicationEventPublisher.publishEvent(new FailedPBAPaymentEvent(callbackRequest, C2_APPLICATION));
-        }
-
-        C2DocumentBundle c2DocumentBundle = caseData.getLastC2DocumentBundle();
-
+        final C2DocumentBundle c2DocumentBundle = caseData.getLastC2DocumentBundle();
         applicationEventPublisher.publishEvent(new C2UploadedEvent(callbackRequest, c2DocumentBundle));
+
         if (isNotPaidByPba(c2DocumentBundle)) {
+            log.info("C2 payment for case {} not taken due to user decision", caseDetails.getId());
             applicationEventPublisher.publishEvent(new C2PbaPaymentNotTakenEvent(callbackRequest));
+        } else {
+            if (displayAmountToPay(caseDetails)) {
+                try {
+                    paymentService.makePaymentForC2(caseDetails.getId(), caseData);
+                } catch (FeeRegisterException | PaymentsApiException paymentException) {
+                    log.error("C2 payment for case {} failed", caseDetails.getId());
+                    applicationEventPublisher.publishEvent(new FailedPBAPaymentEvent(callbackRequest, C2_APPLICATION));
+                }
+            } else if (NO.getValue().equals(caseDetails.getData().get(DISPLAY_AMOUNT_TO_PAY))) {
+                log.warn("C2 payment for case {} not taken as payment fee not shown to user", caseDetails.getId());
```

Comment:
Just thinking, shouldn't this be an error as well? I believe the warns appear only in a trace, so we could potentially miss it.